src/util/templates/GenericTemplateEngine.ts

```diff
@@ -0,0 +1,17 @@
+import type { AsyncHandler } from '../handlers/AsyncHandler';
+import type { Template, TemplateEngineInput } from './TemplateEngine';
+import { TemplateEngineHandler } from './TemplateEngineHandler';
+import Dict = NodeJS.Dict;
+
+export class GenericTemplateEngine<T extends Dict<any> = Dict<any>> extends TemplateEngineHandler<T> {
```

Comment:
I just made several comments about renaming stuff because we there is now a class called `GenericTemplateEngine`, and in this comment I'm going to recommend renaming it 😅 (the previous comments still stand though, don't think the word "Generic" adds much).

Since there is now only 1 place where the hbs/ejs template engines get defined, and every time a fixed template gets used I would suggest some changes that simplifies the classes. `EjsTemplateEngine` and `HandleBarsTemplateEngine` should not have an optional `template` parameter. We never use it anymore, and probably never will, due to the changes.

I would then also rename `GenericTemplateEngine` to `StaticTemplateEngine` and have `template` be a required parameter. The point of this class is then for cases where we know the template in advance. We could even throw an error in case its function ever gets called with a `template` parameter, not sure if there currently is any class that depends on the fact that this is allowed.